### ISSUE TITLE
relocation R_X86_64_PC32 against protected symbol `common' can not be used when making a shared object

### ISSUE THREAD

**Issue by xsellier** [obsproject/obs-studio#5622]
### Operating System Info
Debian

### Other OS

Linux binogure 5.15.0-2-amd64 #1 SMP Debian 5.15.5-1 (2021-11-26) x86_64 GNU/Linux

### OBS Studio Version

Git tag: 27.1.3

### OBS Studio Compilation Log URL

https://pastebin.com/fqQ2Ds0d

### Expected Behavior

COMPILATION DONE

### Current Behavior

```
/usr/bin/ld: CMakeFiles/obs-filters.dir/rnnoise/src/denoise.c.o: relocation R_X86_64_PC32 against protected symbol `common' can not be used when making a shared object
/usr/bin/ld: final link failed: bad value
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

### Steps to Reproduce

Running this script (follow build instructions: <https://github.com/obsproject/obs-studio/wiki/install-instructions#debian-based-build-directions>):
```shell
sudo apt-get install build-essential cmake git libmbedtls-dev libasound2-dev libavcodec-dev libavdevice-dev libavfilter-dev libavformat-dev libavutil-dev libcurl4-openssl-dev \
 libfdk-aac-dev libfontconfig-dev libfreetype6-dev libglvnd-dev libjack-jackd2-dev libjansson-dev libluajit-5.1-dev libpulse-dev libqt5x11extras5-dev libspeexdsp-dev libswresample-dev libswscale-dev libudev-dev libv4l-dev \
 libvlc-dev libwayland-dev libx11-dev libx264-dev libxcb-shm0-dev libxcb-xinerama0-dev libxcomposite-dev libxinerama-dev pkg-config python3-dev qtbase5-dev qtbase5-private-dev libqt5svg5-dev swig \
 libxcb-randr0-dev libxcb-xfixes0-dev libx11-xcb-dev libxcb1-dev libxss-dev qtwayland5 libgles2-mesa libgles2-mesa-dev
sudo apt install libnss3-dev
wget https://cdn-fastly.obsproject.com/downloads/cef_binary_4280_linux64.tar.bz2
tar -xjf ./cef_binary_4280_linux64.tar.bz2
git clone --recursive https://github.com/obsproject/obs-studio.git
cd obs-studio
mkdir build && cd build
# Note Ubuntu 20.04/Debian 10 must set ENABLE_PIPEWIRE=OFF and do not support wayland capture.
# Modern platforms can use the default/enable pipewire for wayland capture support.
cmake -DUNIX_STRUCTURE=1 -DENABLE_PIPEWIRE=OFF -DCMAKE_INSTALL_PREFIX=/usr -DBUILD_BROWSER=ON -DCEF_ROOT_DIR="../../cef_binary_4280_linux64" ..
make -j$(nproc)
```

**Comment by xsellier** [comment]
Found a workaround:
```shell
git clone https://github.com/xiph/rnnoise.git
cd rnnoise/
./autogen.sh
./configure
make
sudo checkinstall --default --pkgname=rrnoise  --fstrans=no --backup=no --pkgversion="$(date +%Y%m%d)-git" --deldoc=yes
```

Then you can compile OBS-Studio:
```shell
sudo apt-get install build-essential cmake git libmbedtls-dev libasound2-dev libavcodec-dev libavdevice-dev libavfilter-dev libavformat-dev libavutil-dev libcurl4-openssl-dev \
 libfdk-aac-dev libfontconfig-dev libfreetype6-dev libglvnd-dev libjack-jackd2-dev libjansson-dev libluajit-5.1-dev libpulse-dev libqt5x11extras5-dev libspeexdsp-dev libswresample-dev libswscale-dev libudev-dev libv4l-dev \
 libvlc-dev libwayland-dev libx11-dev libx264-dev libxcb-shm0-dev libxcb-xinerama0-dev libxcomposite-dev libxinerama-dev pkg-config python3-dev qtbase5-dev qtbase5-private-dev libqt5svg5-dev swig \
 libxcb-randr0-dev libxcb-xfixes0-dev libx11-xcb-dev libxcb1-dev libxss-dev qtwayland5 libgles2-mesa libgles2-mesa-dev
sudo apt install libnss3-dev
wget https://cdn-fastly.obsproject.com/downloads/cef_binary_4280_linux64.tar.bz2
tar -xjf ./cef_binary_4280_linux64.tar.bz2
git clone --recursive https://github.com/obsproject/obs-studio.git
cd obs-studio
mkdir build && cd build
# Note Ubuntu 20.04/Debian 10 must set ENABLE_PIPEWIRE=OFF and do not support wayland capture.
# Modern platforms can use the default/enable pipewire for wayland capture support.
cmake -DUNIX_STRUCTURE=1 -DENABLE_PIPEWIRE=OFF -DCMAKE_INSTALL_PREFIX=/usr -DBUILD_BROWSER=ON -DCEF_ROOT_DIR="../../cef_binary_4280_linux64" ..
make -j$(nproc)
sudo checkinstall --default --pkgname=obs-studio --fstrans=no --backup=no --pkgversion="$(date +%Y%m%d)-git" --deldoc=yes
```

**Comment by kkartaltepe** [comment]
This is an issue with clang, you can use gcc to work around it.

**Comment by kkartaltepe** [comment]
This is a good first issue, you can convert the rnnoise cmake file to use hidden visibility by default which should prevent this symbol issue (and is how upstream is compiled). We have an example of this in plugins/obs-outputs/CMakeLists.txt:98 (this is for CXX, but rnnoise needs this for C)

**Comment by kkartaltepe** [comment]
Thanks looks like it was fixed in #6226
